contracts/tokens/NFTokenEnumerable.sol

```diff
@@ -52,6 +52,29 @@ contract NFTokenEnumerable is NFToken, ERC721Enumerable {
     tokens.push(_tokenId);
   }
 
+  /*
+   * @dev Burns a NFToken.
+   * @param _owner Address of the NFToken owner.
+   * @param _tokenId ID of the NFToken to be burned.
+   */
+  function _burn(address _owner,
+                 uint256 _tokenId)
+    internal
+  {
+    super._burn(_owner, _tokenId);
+
+    uint256 tokenIndex = idToIndex[_tokenId];
+    uint256 lastTokenIndex = tokens.length.sub(1);
```

Comment:
Can we add `assert(tokens.length > 0);` here?

Reply:
Will do.